apps/dashboard/app/models/script.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class Script
+  include ActiveModel::Model
+
+  attr_reader :title
+
+  attr_reader :id
+
+  class << self
+
+    def find(id, project_dir)
+      all(project_dir).select do |script|
+        # Rails.logger.debug("#{id} == #{script.id}")
+        id == script.id
+      end.first
+    end
+
+    def all(project_dir)
+      Dir.glob("#{Project.dataroot}/#{project_dir}/.ondemand/scripts/*.yml").map do |file|
+        Script.from_yaml(file, project_dir)
+      end
+    end
+
+    def from_yaml(file, project_dir)
+      contents = File.read(file)
+      opts = YAML.safe_load(contents).to_h
+      opts.merge!({ id: File.basename(file, '.yml') })
+      opts.merge!({ project_dir: project_dir.to_s })
+      new(opts)
+    end
+    alias_method :from_yml, :from_yaml
+
+    def next_id(project_dir)
+      all(project_dir).map(&:id).map(&:to_i).max + 1
+    end
+  end
+
+  def initialize(opts = {})
+    opts = opts.to_h.with_indifferent_access
+
+    @project_dir = opts[:project_dir] || raise(StandardError, 'You must set the project directory')
+    @id = opts[:id]
+    @title = opts[:title].to_s
+  end
+
+  def to_yaml
+    { 'title' => title }.to_yaml
+  end
+  alias_method :to_yml, :to_yaml
+
+  def to_h
+    {}.tap do |hsh|
+      instance_variables.each do |var|
+        hsh[var] = instance_variable_get(var)
+      end
+    end
+  end
```

Comment:
I need some help here. What is the use of this and why have it?

Reply:
It's an alias for `inspect`, for debugging. Sometimes when I'm debugging I like to do something like 

```ruby
raise StandardError, the_thing_im_working_with.inspect
```
to see what the object I'm interacting with and what it's state is.

Reply:
Ok, does that mean it will eventually get removed or is it going to stay in the code?

Reply:
> Ok, does that mean it will eventually get removed or is it going to stay in the code?

I figured we'd keep it, I mean we'll always need to debug right?